portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -73,20 +73,42 @@ public void doGet(HttpServletRequest request, HttpServletResponse response) thro
     loadCommentData(request);
 
     int requestAmount;
+    int pageNumber;
     try {
       requestAmount = Integer.parseInt(request.getParameter("amount"));
     } catch (NumberFormatException e) {
       requestAmount = DEFAULT_COMMENTS;
     }
+
+    try {
+      pageNumber = Integer.parseInt(request.getParameter("page"));
+    } catch (NumberFormatException e) {
+      pageNumber = 0;
+    }
     if (requestAmount < 0) {
       requestAmount = DEFAULT_COMMENTS;
     }
     // Reduce request size to be the comments size to not get exception
     if (requestAmount > comments.size()) {
       requestAmount = comments.size();
     }
+
+    int offsetAmount = pageNumber * requestAmount;
+    if (offsetAmount < 0) {
+      offsetAmount = 0;
+    }
+    if (offsetAmount > comments.size()) {
+      offsetAmount = comments.size();
+    }
+
```

Comment:
this can be merged to "if (requestAmount + offsetAmount > comments.size())"

if (requestAmount + offsetAmount > comments.size()) {
    offsetAmount = **
    requestAmount = **
}

Reply:
Will resolve in my next pull request